src/goesproc/config.cc

```diff
@@ -111,6 +112,99 @@ bool loadHandlers(const toml::Value& v, Config& out) {
       }
     }
 
+    auto gradient = th->find("gradient");
+    if (gradient) {
+      auto trs = gradient->as<toml::Table>();
+      for (const auto& it : trs) {
+        auto interpolation = it.second.find("interpolation");
```

Comment:
Can you add an `auto channel = toUpper(it.first);` here, so you can use `channel` below.